test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2432,17 +2439,34 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
         }
     }
 
-    void test_am_send_recv(size_t size, size_t hdr_size = 0ul)
+    void test_am_send_recv(size_t size, size_t hdr_size = 0ul,
+                           size_t num_iters = m_num_iters,
+                           bool recv_prereg_memh = false,
+                           uct_md_h *recv_mds = NULL)
     {
         /* send multiple messages to test the protocol both before and after
          * connection establishment */
-        for (int i = 0; i < m_num_iters; i++) {
+        for (int i = 0; i < num_iters; i++) {
             std::string sb(size, 'x');
             std::string rb(size, 'y');
             std::string shdr(hdr_size, 'x');
             std::string rhdr(hdr_size, 'y');
+            ucp_mem_h rmemh(NULL);
+
+            if (recv_prereg_memh) {
+                rmemh = receiver().mem_map(&rb[0], size);
+                if (recv_mds != NULL) {
```

Comment:
1. why here and not in the test itself?
2. can this be outside of `if (recv_prereg_memh) {` ?